src/custom-elements/src/components/spinner/spinner.tsx

```diff
@@ -0,0 +1,75 @@
+import { Component, Prop, Watch } from '@stencil/core';
+import { ICON_SIZES } from '../../constants/size';
+
+const SPINNER_COLORS = ['', 'primary', 'success', 'warning', 'danger', 'muted', 'secondary', 'light'];
+
+@Component({
+  tag: 'chi-spinner',
+  styleUrl: 'spinner.scss',
+  scoped: true
+})
+export class Spinner {
+  /**
+   *  to set size of a spinner { xs, sm, sm--2, sm--3, md, lg, xl, xxl }.
+   */
+  @Prop({ reflectToAttr: true }) size: string;
+
+  /**
+   *  to set color of a spinner { primary, success, warning, danger, muted }.
+   */
+  @Prop({ reflectToAttr: true }) color: string;
+
+  /**
+   *  to render spinners on dark or light backgrounds.
+   */
+  @Prop({ reflectToAttr: true }) backdrop: string;
+
+  @Watch('size')
+  sizeValidation(newValue: string) {
+    if (newValue && !ICON_SIZES.includes(newValue)) {
+      throw new Error(`${newValue} is not a valid size for spinner. Valid values are xs, sm, sm--2, sm--3, md, lg, xl, xxl or ''. `);
+    }
+  }
+
+  @Watch('color')
+  colorValidation(newValue: string) {
+    if (newValue && !SPINNER_COLORS.includes(newValue)) {
+      throw new Error(`${newValue} is not a valid color for spinner. Valid values are primary, success, warning, danger, muted, secondary, light or ''. `);
+    }
+  }
+
+  @Watch('backdrop')
+  backdropValidation(newValue: string) {
+    if (newValue && newValue !== 'inverse' && newValue !== 'backdrop') {
+      throw new Error(`${newValue} is not a valid backdrop for spinner. Valid values are inverse, backdrop, '' or true. `);
```

Comment:
@SergioQCL in this case true is a string instead of boolean, so for the user it will feel like he sets the backdrop to true. Empty string plays role when user just uses the attribute name and no value assigned to it, If you observe in the docs you will see backdrop doesn't need any value to be set.

Reply:
Just to clarify. Could you tell me, please, the differences between using true and backdrop as values?
And using the empty string will render no backdrop, won't it?

Reply:
@SergioQCL ```backdrop='backdrop'``` ```backdrop='true'``` ```backdrop``` all three will render a backdrop

Reply:
@SergioQCL If an attribute is boolean, the presence of it will be true and the omission will be false. Even backdrop='false' will render a backdrop. Srinath configured the backdrop as a string, so if the developer omits the attribute there won't be any backdrop, if it is set to inverse, it will be inverse and any other value will set a default backdrop.

Reply:
@srinath-koneru but that validation won't take 'true' or '' as correct values, will it?

Reply:
@SergioQCL it won't take 'true' as correct value but '' is, if we set ```backdrop``` on chi-spinner it's nothing but ```backdrop=''```. But 'true' is not getting used I'll add that, what a bummer. Thanks